packages/ozone/src/tag-service/index.ts

```diff
@@ -0,0 +1,61 @@
+import { ModerationService } from '../mod-service'
+import { ModSubject } from '../mod-service/subject'
+import { langLogger as log } from '../logger'
+import { ContentTagger } from './content-tagger'
+import { LanguageTagger } from './language-tagger'
+import { EmbedContentTypeTagger } from './embed-content-type-tagger'
+import { ModerationSubjectStatusRow } from '../mod-service/types'
+
+export class TagService {
+  private taggers: ContentTagger[]
+
+  constructor(
+    private subject: ModSubject,
+    protected subjectStatus: ModerationSubjectStatusRow | null,
+    private taggerDid: string,
+    private moderationService: ModerationService,
+  ) {
+    this.taggers = [
+      new LanguageTagger(subject, subjectStatus, moderationService),
+      new EmbedContentTypeTagger(subject, subjectStatus, moderationService),
+      // Add more taggers as needed
+    ]
+  }
+
+  async evaluateForSubject() {
+    try {
+      const tags: string[] = []
+
+      await Promise.all(
+        this.taggers.map(async (tagger) => {
+          try {
+            if (tagger.isApplicable()) {
+              const newTags = await tagger.getTags()
+              if (newTags.length) tags.push(...newTags)
+            }
```

Comment:
What is the intended behavior if you call `tagger.getTags()` without checking `tagger.isApplicable()`?  It seems like `isApplicable()` may not need to be part of the public interface— the implementation of `getTags()` could alternatively check this itself and no-op by returning `[]`.

Reply:
I normally try to split these and let the consumer of the class have the option to perform necessary checks before invoking the getter. for getters like this, i often find use-cases down the line where we would wanna know "what would be the tags IF they were applicable?". 
however, this isn't the case here... yet so I'm happy to make that check implicit.

Reply:
That makes sense—it just seemed like the class could be prone to misuse, as it seemed implementations of `getTags()` were assuming it was gated by `isApplicable()`.

I think there's a way to get the best of both worlds using the abstract class, where there's a default implementation for `getTags()` that performs the `isApplicable()` check.  And then there would be a separate method for implementers `getTagsImple()` that is used within the default implementation of `getTags()`.

Reply:
ohh I see what you mean. refactored it a bit to get that going.